proxy/src/registry.rs

```diff
@@ -624,6 +648,31 @@ impl Client for Registry {
         ))
     }
 
+    async fn unregister_user(
+        &self,
+        author: &protocol::ed25519::Pair,
+        handle: Id,
+        fee: Balance,
+    ) -> Result<Transaction, error::Error> {
+        // Prepare and submit user unregistration transaction.
+        let unregister_message = protocol::message::UnregisterUser {
```

Comment:
Heh... So _all_ these functions look the same! Recalling back to @MeBrei's PR yesterday. I'm _highly_ interested to see if we could find the right abstraction :eyes:

Reply:
Yeah, agree. Could you maybe move that discussion into a dedicated issue?

Reply:
#672